plat/kvm/arm/setup.c

```diff
@@ -159,6 +159,36 @@ static void __noreturn _ukplat_entry2(void)
 	ukplat_lcpu_halt();
 }
 
+#if CONFIG_LIBUKRSI && CONFIG_EARLY_UART_NS16550
+static int uk_rsi_setup_early_uart(struct ukplat_bootinfo *bi)
+{
+	int rc;
+	struct ukplat_memregion_desc mrd = {0};
+
+	__u64 aliased_IPA = CONFIG_EARLY_UART_NS16550_BASE;
+	__sz uart_mmio_len = __PAGE_SIZE; /* page-aligned */
+
+	/* Add a memreg for the Unprotected IPA alias of the early
+	 * UART to prevent from unmapping at paging_init().
+	 */
+	mrd.vbase = (__vaddr_t)aliased_IPA;
+	mrd.pbase = (__paddr_t)aliased_IPA;
+	mrd.len   = uart_mmio_len;
+	mrd.type  = UKPLAT_MEMRT_REALM;
+	mrd.flags = UKPLAT_MEMRF_READ | UKPLAT_MEMRF_WRITE | UKPLAT_MEMRF_MAP;
+
+	rc = ukplat_memregion_list_insert(&bi->mrds, &mrd);
+	if (unlikely(rc < 0)) {
+		uk_pr_err("Could not add free memory descriptor");
+		return rc;
+	}
+
+	ukplat_memregion_list_coalesce(&bi->mrds);
```

Comment:
It's not a good idea to ignore error code of `ukplat_memregion_list_coalesce()`. You can have this function return that.